ntp-proto/src/filter.rs

```diff
@@ -393,8 +419,10 @@ fn find_interval(chime_list: &[CandidateTuple]) -> (NtpDuration, NtpDuration) {
         chime = 0;
         for tuple in chime_list.iter().rev() {
             chime += tuple.endpoint_type as i32;
-            if chime >= (n - found) as i32 {
-                high = tuple.edge;
+
+            // the code skeleton uses `n - found` here, which is wrong!
+            if chime >= (n - allow) as i32 {
+                high = Some(tuple.edge);
                 break;
```

Comment:
why can't you break from a `for`/`while` loop with a value? Seems to me like that should just work.

Reply:
Because `for`/`while` might break out of the loop due to their iterator running out/the loop condition becoming false, in which case you wouldn't have the opportunity to yield a value.

Reply:
ah right, you need something like python's `for`/`else` construct